### PR TITLE
Add mint::freeze_authority attribute when init-ing a mint.

### PR DESCRIPTION
**Pull request by cqfd**
This came up in discord and I thought it would be a good excuse to learn some more macro stuff (no worries if this isn't actually a feature that needs to be included in anchor!).

### CODE REVIEW

File: lang/syn/src/lib.rs [hunk 1]
```diff
@@ -661,9 +662,19 @@ pub enum InitKind {
...
+    },
+    AssociatedToken {
+        owner: Expr,
+        mint: Expr,
+    },
+    Mint {
+        owner: Expr,
+        freeze_authority: Option<Expr>,
+        decimals: Expr,
+    },
```
Comment (cqfd) [inline lang/syn/src/lib.rs:677]: Not sure why my rust fmt formats this different :/

**Comment by armaniferrante** [comment]
> This came up in discord and I thought it would be a good excuse to learn some more macro stuff (no worries if this isn't actually a feature that needs to be included in anchor!).

This is very useful!

File: lang/syn/src/lib.rs (hunk 1 above)
Comment (jpbogle) [reply lang/syn/src/lib.rs:679]: you mean the order of the fields? or just the indentation

**Review by jpbogle — COMMENTED** [review]
cool thanks @cqfd this is exactly what i was thinking. Was going to make a PR but i wasnt sure all the places i would need to look. This is a good example for me to see

File: lang/syn/src/lib.rs (hunk 1 above)
Comment (cqfd) [reply lang/syn/src/lib.rs:679]: Yeah just indentation stuff—it added newlines to `Token` and `AssociatedToken` that I wasn't expecting (my change only touches `Mint`).

**Pull request merged**